### PR TITLE
Add Route::all and Route::nest

### PR DESCRIPTION
**Pull request by tirr-c**
Follow-up of #364.

`Route::all` catches all HTTP methods. This is mandatory for nesting story, but I missed this somehow.

This PR also adds a convenience method `nest_service`, which turns on nesting flag temporarily and calls `all`.

### CODE REVIEW

File: src/server/route.rs [hunk 1]
```diff
@@ -62,6 +64,19 @@ impl<'a, State: 'static> Route<'a, State> {
         self
     }
 
+    /// Nest a [`Server`] at the current path.
+    ///
+    /// [`Server`]: struct.Server.html
+    pub fn nest_service(&mut self, service: crate::Server<State>) -> &mut Self
```
Comment (yoshuawuyts) [inline src/server/route.rs]: I wonder if we could shorten this to `nest`. I'm still wondering if we could implement `Service` on `Server` directly, at which point the word `service` will likely disappear from most places. `nest` still feels quite clear to me.

File: tests/nested.rs [hunk 2]
```diff
@@ -53,7 +50,7 @@ fn nested_middleware() {
         app.middleware(test_middleware);
         app.at("/echo").get(echo_path);
         app.at("/:foo/bar").strip_prefix().get(echo_path);
-        route.strip_prefix().get(app.into_http_service());
```
Comment (yoshuawuyts) [inline tests/nested.rs]: Should we remove the `strip_prefix` method too? Or are there any other uses left for it still?

File: src/server/route.rs (hunk 1 above)
Comment (tirr-c) [reply src/server/route.rs]: Well, I think it's okay to leave this as is, because `tide::Server` is more like a builder type for `tide::Service`. This seems natural for me, how do you think?

And yeah, `nest` seems clear enough; maybe we can remove existing method and rename this to `nest`?

File: tests/nested.rs (hunk 2 above)
Comment (tirr-c) [reply tests/nested.rs]: I left the method for `HttpService`s other than Tide. Users might want to nest some service and call `route.strip_prefix().all(my_http_service)`.

Comment (yoshuawuyts) [reply tests/nested.rs]: Alright, fair enough!

File: src/server/route.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/server/route.rs]: Renaming to `nest` sounds good.

Also I've been wondering if we need a distinction between `Server` and `Service`. It seems that if they could be the same struct that would allow us to consolidate the API a bit. But that's something for the future.

**Comment by tirr-c** [comment]
Renamed the method, and merging!
